src/highdicom/map/sop.py

```diff
@@ -0,0 +1,789 @@
+                    'floating-point types are supported.'
+                )
+        elif pixel_array.dtype.kind == 'u':
+            if pixel_array.dtype not in (np.uint8, np.uint16):
+                raise ValueError(
+                    'Unsupported unsigned integer type for pixel data: '
+                    '16-bit unsigned integer types are supported.'
+                )
+            return (
+                _PixelDataType.USHORT,
+                self._pixel_data_type_map[_PixelDataType.USHORT],
+            )
+        elif pixel_array.dtype.kind == "i":
+            if pixel_array.dtype not in (np.int8, np.int16):
+                raise ValueError(
+                    'Unsupported signed integer type for pixel data: '
+                    '8-bit or 16-bit signed integer types are supported.'
+                )
+            return (
+                _PixelDataType.SHORT,
+                self._pixel_data_type_map[_PixelDataType.SHORT],
+            )
+        raise ValueError(
+            'Unsupported data type for pixel data.'
+            'Supported are 8-bit or 16-bit signed and unsigned integer types '
+            'as well as 32-bit (single-precision) or 64-bit (double-precision) '
+            'floating-point types.'
+        )
+
+    def _encode_pixels(self, plane: np.ndarray) -> bytes:
+        """Encodes a given pixel array as a bytes object
+
+        Parameters
+        ----------
+        plane : np.ndarray
+            The numpy array to encode
+
+        Returns
+        -------
+        bytes
+            `plane` encoded as a `bytes` object
+
+        Raises
+        ------
+        ValueError
+            If the SOP instance uses an encapsulated transfer syntax and
+            `plane` is not exactly 2 dimensional.
+        """
+        if self.file_meta.TransferSyntaxUID.is_encapsulated:
+            # Check that only a single plane was passed
+            if plane.ndim != 2:
+                raise ValueError(
+                    'Only single frame can be encoded at at time '
+                    'in case of encapsulated format encoding.'
+                )
+            return encode_frame(
+                plane.astype(np.uint16),
+                transfer_syntax_uid=self.file_meta.TransferSyntaxUID,
+                bits_allocated=self.BitsAllocated,
+                bits_stored=self.BitsStored,
+                photometric_interpretation=self.PhotometricInterpretation,
+                pixel_representation=self.PixelRepresentation,
+            )
+        else:
+            if plane.dtype == np.uint8:
+                return plane.astype(np.uint16).flatten().tobytes()
+            elif plane.dtype.kind == 'i':
+                plane = plane.astype(np.int16) + 2 ** 16 / 2
+                return plane.astype(np.uint16).flatten().tobytes()
+            else:
+                return plane.flatten().tobytes()
+    
+    @property
+    def pixel_array(self) -> np.ndarray:
```

Comment:
@hackermd here's what I came up with as a workaround for the PixelRepresentation bug in pydicom, let me know what you think.

Reply:
Yes, I saw that. That was quite an elegant solution. However, the problem is that the Pixel Data element in the `pydicom.Dataset` could not be decoded when it would be read by `pydicom.dcmread()`. Including the attribute into the data set is also not ideal, though. I will report this to pydicom..

Reply:
Here is the pydicom issue: https://github.com/pydicom/pydicom/issues/1457

Reply:
Should get merged into pydicom soon: https://github.com/pydicom/pydicom/pull/1458